ui/shared/GasSelector.qml

```diff
@@ -0,0 +1,181 @@
+import QtQuick 2.13
+import QtQuick.Controls 2.13
+import QtQuick.Layouts 1.13
+import "../imports"
+import "./"
+
+Item {
+    id: root
+    anchors.left: parent.left
+    anchors.right: parent.right
+    height: gasSlider.height + Style.current.smallPadding * 3 + txtNetworkFee.height + buttonAdvanced.height
+    property double slowestGasPrice: 0
+    property double fastestGasPrice: 100
+    property string gasLimit: ""
+    property double stepSize: ((root.fastestGasPrice - root.slowestGasPrice) / 10).toFixed(1)
+    property alias value: gasSlider.value
+    property var getGasEthValue: function () {}
+    property string defaultCurrency: "USD"
+    property string selectedGasPrice: ""
+    property string selectedGasLimit: ""
+
+    function resetGasSlider() {
+        return ((50 * (root.fastestGasPrice - root.slowestGasPrice) / 100) + root.slowestGasPrice)
+    }
+
+    StyledText {
+        id: txtNetworkFee
+        anchors.top: parent.top
+        anchors.left: parent.left
+        text: qsTr("Network fee")
+        font.weight: Font.Medium
+        font.pixelSize: 13
+        color: Style.current.textColor
+    }
+
+    StyledText {
+        id: labelGasPriceSummary
+        anchors.top: parent.top
+        anchors.right: labelEth.left
+        text: "0.0"
+        font.weight: Font.Medium
+        font.pixelSize: 13
+        color: Style.current.secondaryText
+    }
+
+    StyledText {
+        id: labelEth
+        anchors.top: parent.top
+        anchors.right: parent.right
+        text: " ETH"
+        font.weight: Font.Medium
+        font.pixelSize: 13
+        color: Style.current.secondaryText
+    }
+
+
+    Item {
+        id: sliderWrapper
+        anchors.topMargin: Style.current.smallPadding
+        anchors.top: labelGasPriceSummary.bottom
+        height: sliderWrapper.visible ? gasSlider.height + labelSlow.height + Style.current.padding : 0
+        width: parent.width
+        visible: root.selectedGasPrice == "" && root.selectedGasLimit == ""
+
+        StatusSlider {
+            id: gasSlider
+            minimumValue: root.slowestGasPrice
+            maximumValue: root.fastestGasPrice
+            stepSize: root.stepSize
+            value: root.resetGasSlider()
+            onValueChanged: {
+                if (!isNaN(gasSlider.value)) {
+                    labelGasPriceSummary.text = root.getGasEthValue(gasSlider.value, root.gasLimit)
```

Comment:
When `root.getGasEthValue()` is called, or in other words, when the `SendModal` in the wallet is opened, the app crashes... I don't really know why. @richard-ramos mind giving this a spin?